python/lsst/sims/maf/metricBundles/metricBundleGroup.py

```diff
@@ -319,21 +318,21 @@ def _runCompatible(self, compatibleList):
                 # There is data! Should we use our data cache?
                 if cache:
                     # Make the data idxs hashable.
-                    cacheKey = str(sorted(slice_i['idxs']))[1:-1].replace(' ','')
+                    cacheKey = frozenset(slice_i['idxs'])
                     # If key exists, set flag to use it, otherwise add it
                     if cacheKey in cacheDict:
                         useCache = True
                     else:
                         cacheDict[cacheKey] = i
                         useCache = False
-                    # If we are above the cache size, drop the oldest element from the cache dict
-                    if i > slicer.cacheSize:
-                        cacheDict.popitem(last=False) #remove 1st item
                     for b in bDict.itervalues():
                         if useCache:
                             b.metricValues.data[i] = b.metricValues.data[cacheDict[cacheKey]]
                         else:
                             b.metricValues.data[i] = b.metric.run(slicedata, slicePoint=slice_i['slicePoint'])
+                    # If we are above the cache size, drop the oldest element from the cache dict
+                    if len(cacheDict) > slicer.cacheSize:
+                        cacheDict.popitem(last=False)
```

Comment:
Great - this is what was causing Eric's problem I think (we were checking if the key existed, but then deleting it before use).